libs/s25main/figures/nofTempleServant.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2005 - 2021 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "nofTempleServant.h"
+#include "GamePlayer.h"
+#include "Loader.h"
+#include "SoundManager.h"
+#include "WineLoader.h"
+#include "buildings/nobUsual.h"
+#include "network/GameClient.h"
+#include "ogl/glArchivItem_Bitmap_Player.h"
+#include "random/Random.h"
+#include "world/GameWorld.h"
+
+namespace {
+int producedGood = 0;
```

Comment:
That has to be an instance variable doesn't it? In case you have multiple servants, which is very likely.

Reply:
Refactored and solved